### PR TITLE
Migrate Terraform from Cross Functional Repo to Infrastructure Repo

### LINKED ISSUE #6: Infrastructure: Move Terraform provisioning pipeline
Currently, the infrastructure provisioning pipeline and terraform lives in the Cross Functional Requirements repository. These files need to be moved to Infrastructure repository and verified that they run correctly.

### Acceptance Criteria
- [x] Ensure that infrastructure provisioning workflow files have been moved to Infrastructure repo
- [x] Ensure that the terraform files have been moved to the Infrastructure repo
- [x] Ensure that terraform service principal's client secrets are available in the workflow
- [x] Ensure that infrastructure is provisioned correctly by the CD pipeline using client secrets
- [x] Ensure that provisioning pipeline **continues** to be executable on dev machine (can run terraform locally)

### LINKED ISSUE #12: Infrastructure: Move Terraform code
Currently, the infrastructure code (terraform) lives in the Cross Functional Requirements repository. We need to move this code to the infrastructure repository so that all of the infrastructure code is in the same place. 

**Note:** To clarify, this issue is to migrate the terraform and not the pipelines, which were handled in a previous issue (#11 #10)

### AC
- [x] All terraform code should be moved from Cross Functional Requirements to Infrastructure repository
- [x] Terraform should deploy infrastructure from local developer machine
- [x] Terraform should deploy via CD pipeline
- [x] Terraform state is stored in the admin storage account (can re-run deployment without errors)

### PR DESCRIPTION
**Pull request by adelechurchill**
Addresses Issue: https://github.com/buerokratt/Infrastructure/issues/12

Acceptance Criteria:
- All terraform code should be moved from Cross Functional Requirements to Infrastructure repository
  * Existing code copied and manually tested for functionality
- Terraform should deploy infrastructure from local developer machine
   * TF continues to deploy from local machine 
- Terraform should deploy via CD pipeline
  * GH Workflow executes as intended and uses GH secrets to deploy to a specific Azure Sub
- Terraform state is stored in the admin storage account (can re-run deployment without errors)
  * State is stored in admin storage account with corresponding container allowing re-provisioning of resources from tfstate file

Not included in this PR and will be covered in separate PR:
- Add pipeline to setup project pre-requisite resources via azure cli
- Store kubectl config in GitHub secrets for infrastructure repo

### CODE REVIEW

File: infrastructure/Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		terraform init && \
+		terraform apply -auto-approve -var-file=./environments/dev.tfvars -var-file=credentials.tfvars -lock=false
+
+tf-destroy:
+	@cd $(tf_development) && \
+		terraform destroy -var-file=./environments/dev.tfvars -lock=false
+
+tf-format:
+	terraform fmt -recursive $(tf_development)
+
```
Comment (guypritchard) [inline infrastructure/Makefile:28]: Worth a comment to indicate why this is commented out at the moment? For future humans?

File: infrastructure/modules/resource_templates/aks/main.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+data "azurerm_resource_group" "resource_group" { name = var.resource_group_name }
+
+resource "azurerm_kubernetes_cluster" "aks" {
+  name                = var.aks_name
+  location            = data.azurerm_resource_group.resource_group.location
+  resource_group_name = data.azurerm_resource_group.resource_group.name
+  dns_prefix          = var.aks_name
+
+  default_node_pool {
+    name       = "default"
```
Comment (guypritchard) [inline infrastructure/modules/resource_templates/aks/main.tf:10]: Perhaps not for now - but this feels like environment specific config or a default variable perhaps.

File: infrastructure/versions.tf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.5.0"
+    }
+  }
+  required_version = ">= 1.1.0"
+
+  backend "azurerm" {}
+  #backend "local" {
```
Comment (guypritchard) [inline infrastructure/versions.tf:12]: If we want to keep this commented out - perhaps a quick one-liner to indicate its purpose?

**Review by guypritchard — APPROVED** [review]
A few minor comments - nice work!

File: infrastructure/modules/resource_templates/aks/main.tf (hunk 2 above)
Comment (Adele-Churchill-13) [reply infrastructure/modules/resource_templates/aks/main.tf:10]: Indeed you are correct, this infra is just baseline for now but one to keep in mind when we start to abstract the different environments

**Review by decodingahmed — APPROVED** [review]

File: infrastructure/Makefile (hunk 1 above)
Comment (decodingahmed) [reply infrastructure/Makefile:26]: Done

File: docs/technical/setup-local-dev-env.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Setting up local development environment
+
+Prerequisite Technology Installs
+- Chocolatey
```
[the remark this answers is not in the document]
Comment (decodingahmed) [reply docs/technical/setup-local-dev-env.md:4]: I've captured this in a ticket #21

File: infrastructure/modules/resource_templates/aks/main.tf (hunk 2 above)
Comment (decodingahmed) [reply infrastructure/modules/resource_templates/aks/main.tf:10]: @guypritchard you'll see some more configuration in an upcoming PR very soon

**Review by martinkearn — APPROVED** [review]

**Review by joonasroosalung — APPROVED** [review]

**Pull request merged**